report.md

```diff
@@ -64,10 +64,52 @@
 ## Test
 
 ### Test Plan
-[Describe the testing approach and methodologies.]
+This test plan descripes our current idea of the approach, tools and strategies we are going to be using to ensure the quality and stability of the full-stack application developed using the specified tech stack in this repport for the front and backend. 
+#### Testing opbjectives
+- Ensure application functions correctly across all integrated systems (backend, frontend, database)
+- Validate application security features particularly token mangement and authorization (OIDC)
+- Verify the frontend provides a seamless user experience with state management and responsive design
+- Ensure CI/CD pipelin (Github Actions) operates effectiveely, catching bugs early in the build process
+
+#### Test types and strategies
+ -  Unit testing (Backend & frontend) 
+    - Each backend service (Api endpoints, business logic, token management) will be tested individually
+    - Frontend components and utility functions will be tested for correctness using unit testing frameworks like Jest and React Testing Library.
+
+ - Integration testing
+    - Ensure the backend interacts correctly with the PostgreSQL database for CRUD operations.
+    - Verify that the frontend correctly communicates with backend REST APIs
+
+
+ - End-to-end (E2E) testing
+    - Using Cypress or playwright we are able to simulate real user interactions with the app for testing
+    - We will test critical flows such as authentication, CRUD operations and state management scenarios
+
+ - Security testing
+    - The security tests validate access control through Quarkus’s token-based authentication mechanism (OIDC).
+    - Public endpoints (/all) are tested to ensure they are accessible without authentication.
+    - Protected endpoints (/admin, /user) are restricted based on roles, ensuring that:
+        - Admin users have access to both /admin and /all, but are restricted from user-specific resources.
+        - User accounts have access to /user and proper identity checks are performed (e.g., user-specific greetings).
```

Comment:
These endpoints do not exist.

We only have /auth, /hello and /user.

See openAPI or swagger by running the backend in dev mode and checking `localhost:8080/q/openapi` for openAPI spec and `localhost:8080/q/swagger-ui` for swagger UI.